### ISSUE TITLE
Split design guidelines from API docs

### ISSUE THREAD

**Issue by davidknezic** [axa-ch/style-guide#450]
We've now combined the design guidelines and the API docs. We do this by adding our markup description to the component pages, hidden in a *show code* block.

This works, but IMO it's not the best we can do. This is why I propose splitting the API docs from the design guidelines. Having two fundamentally different parts.

These are some reasons for that idea:

* separate search for technical things like classes/plugins/variables from components
* being able to name technical things differently than the components
* releasing at different paces, probably releasing the API more often
* somewhen maybe even making the design guidelines editable by UX, designers and others (CMS)

What do you think about this?

**Comment by davidknezic** [comment]
I'd love to hear your thoughts @sventschui @elessar-ch @digisz @martinoss @muellermike @TheDadi...

**Comment by sventschui** [comment]
Back to the roots. The issue we had when they were separated was lack of Component docs in Design Guides. If we can get around that im :+1:

**Comment by davidknezic** [comment]
Yes, you're right, I remember. I think we also argued that it's better to have everyone look at the same docs, so every discipline gets some information from the others. But I guess we didn't think about documenting our CSS, JS and assets in more detail. (which is totally necessary, let's say for developers rebuilding our plugins in another technology)

**Comment by muellermike** [comment]
I wouldn't go for that. I guess UX designers or whoever don't care if there are technical things in the search result, do they? And if they are annoyed of the source code block, make this block hidden by default.
Further, you always have to edit both docs (API and design) when you changed a component. Which is leading to different versions on each doc, not the same features in both docs and so on..
We already had situations where the one and only doc of the styleguide wasn't even up to date. So I don't know if it's going to work when there are two docs.
And the biggest point for me: I would rather focus on implementing features, fix bugs, finally bring version 1, then implement this - IMO - not very necessary splitting thing..

So cut a long story short:
I wouldn't rate the benefits we're getting from splitting the docs higher then implementing real features we're using to build our awesome apps. :smile:

**Comment by TheDadi** [comment]
I would go for that approach, if we change the way the docs are generated and the way our plugins work. At the moment it's really akward to find the right markup for the current plugin and the right markup for the right less and so on. Im not happy that our jq plugins or whatever we use in the future are dependent on a specific markup and a specific less elsewhere they wil not work at all. An other thing is that if we change some markup it's not automatically in the docs, someone has to rewrite the example and it's not auto generated. Another thing i want to talk about is independent versioning  of each component and a better change doc when reworking some key things like the grid. 

And the short summary: 
If we do that there are many things to do first e.g. Firstly document all the things we have, then there are still missing some key docs for components and then rework the versioning and build of the whole thingy! But if we do all that, i would go for it. 👍

**Comment by muellermike** [comment]
I agree with you @TheDadi in most of your points. :wink: The quality of the docs should be increased enormously. Including your points like put markup changes automatically in the docs, etc.

My point is just about the splitting thing. Splitting the docs neither brings us those benefits @TheDadi is talking about. So better implement those benefits than split the doc.
And of course - as I mentioned before - the quality of the doc should be increased (I edited my post before and added that point) :wink:

**Comment by davidknezic** [comment]
Thank you @muellermike @TheDadi for the feedback. I think you both have very valid points. Let me propose the following:

* [ ] We first add the most important missing parts to the style guide, including the components we've built with axa.ch. Maybe other projects would also want to add components they've built.
* [ ] We make sure the docs are valid and up to date in todays structure. (no detailed description of the classes available, just markup and a js api)
* [ ] When (almost) everyone is happy, we release 1.x
* [ ] We reconsider how we version things. We've actually solved that very problem in axa.ch, by appending numbers to our BEM blocks (e.g. `button2`).
* [ ] We ask ourselves whether we're happy with the docs, whether we find all the necessary information about BEM classes, states, JS and markup.
* [ ] We throw a party :tada: (actually we should do that much earlier)

**Comment by sventschui** [comment]
`button2`, are you serious? Keeping `button` in the source so one can upgrade without issues to use a new component? Will we also keep the docs for v1?

**Comment by davidknezic** [comment]
There's nothing wrong with that. It doesn't look very nice but works very well and allows us to upgrade much more easily. We don't write any docs for our components, so haven't had that problem yet.

**Comment by sventschui** [comment]
Btw. atlassian does the same:

![image](https://cloud.githubusercontent.com/assets/512692/13770351/ad0107ba-ea86-11e5-9f9e-4cc986f9ff76.png)
